src/main/java/org/opensearch/ad/model/AnomalyResult.java

```diff
@@ -89,42 +98,107 @@
      */
     private final String modelId;
 
-    public AnomalyResult(
-        String detectorId,
-        Double anomalyScore,
-        Double anomalyGrade,
-        Double confidence,
-        List<FeatureData> featureData,
-        Instant dataStartTime,
-        Instant dataEndTime,
-        Instant executionStartTime,
-        Instant executionEndTime,
-        String error,
-        User user,
-        Integer schemaVersion
-    ) {
-        this(
-            detectorId,
-            anomalyScore,
-            anomalyGrade,
-            confidence,
-            featureData,
-            dataStartTime,
-            dataEndTime,
-            executionStartTime,
-            executionEndTime,
-            error,
-            null,
-            user,
-            schemaVersion
-        );
-    }
+    // sequence index (the number of updates to RCF) -- it is possible in imputation
+    // that
+    // the number of updates more than the input tuples seen by the overall program
+    private final Long totalUpdates;
+
+    // flag indicating if the anomaly is the start of an anomaly or part of a run of
+    // anomalies
+    private final Boolean startOfAnomaly;
+
+    // flag indicating if the time stamp is in elevated score region to be
+    // considered as anomaly
+    private final Boolean inHighScoreRegion;
+
+    /**
+     * position of the anomaly vis a vis the current time (can be -ve) if anomaly is
+     * detected late, which can and should happen sometime; for shingle size 1; this
+     * is always 0
```

Comment:
If the shingle size is 4, for example shingle is `[0, 0, 1, 0]`, and this shingle detected as anomaly, but actually the anomaly is caused by the third item "1", then the `relativeIndex` will be 2 ?

Reply:
it'll be -1 (relative to current time) in the library - one can always add (shingleSize - 1) to it to get 2 as well

Reply:
The main goal is to save less result and avoid updating feature's expected value of same interval multiple times. Just some thought, maybe not quite fit the new anomaly detection expected value logic in model.